### DISCUSSION TITLE
Is it possible to "name" a store dynamically at runtime?

### DISCUSSION THREAD

**Discussion by Kristoffer88** [pmndrs/zustand#823] [General]
Hello.

We are using zustand for some apps running in PowerApps model-driven apps.

The same app can be deployed multiple times on the same page (with the same bundle.js, loaded multiple times)

If we have added an app, with zustand loaded two times on the same page, they will share the state between two different bundle.js files.

We have a runtime configuration, that makes the app act differently, so I was hoping, that we could add a name to the store at runtime.

**Comment by AnatoleLucet** [comment]
Hi. If you're speaking about the persist middleware, you can use [`useStore.persist.setOptions`](https://github.com/pmndrs/zustand/wiki/Persisting-the-store's-data#setoptions)

**Comment by eneskabacaoglu** [comment]
[Create Context](https://github.com/pmndrs/zustand#react-context) is exactly for this scenario. I have been in the same situation when I started using Zustand. I have notes component which user can add to a dashboard dynamically and multiple times if they wish. While consuming:

```
import { Provider, createStore } from './BlockStore'

const NoteBlock = ( ) => {
	return (
		<Provider createStore={createStore}>
			<react.Fragment>
				 ........
				 ........
				 ........
			</react.Fragment>
		</Provider>
	)
}
```

While creating (Let's say in BlockStore.js)

```
import create from 'zustand'
import createContext from 'zustand/context'

const { Provider, useStore } = createContext()

const createStore = () =>
	.....
	.....
	.....
export { Provider, useStore, createStore }
```

**Reply by vaynevayne** [reply]
hello @eneskabacaoglu 
It's a little esoteric. Is my understanding correct?
For example, I have a form component and use it in many places
By default, zustand uses one store globally, so all my form will use the same data
When I use provider, then the data of each form is independent, and storage middleware can be used independently. Is that understood this way?
